### DISCUSSION TITLE
Temporary table aliasing

### DISCUSSION THREAD

**Discussion by difof** [go-jet/jet#289] [Q&A] [answered]
Hello. How would I create an alias for temporary table in such query?

```sql
select "id" from (values (1), (2)) as tmptable("id")
```

The goal is combining results of the CTE to be inserted, I have made the jet statement for this query except for the `rewards` CTE which I have no idea how. And my only guess is using `RawStatement` but again, not sure how to drop a `RawStatement` in the `rewards` CTE instead of putting the whole query in a `RawStatement`

```sql
WITH balance_cte AS (
	INSERT INTO md.balances("user_id", "amount")
	VALUES
		(1134, 1000),
		(1178, 1000),
		(1143, 1000)
	RETURNING md.balances."id" AS "tx_id"
), tx_rows AS (
	SELECT "tx_id", ROW_NUMBER() OVER() AS "row_n" FROM balance_cte
), rewards AS (
	SELECT "task_id", ROW_NUMBER() OVER() AS "row_n" FROM
	(VALUES (355), (356), (357))
	AS tasks("task_id")
)
INSERT INTO md.user_rewards ("tx_id", "task_id")
SELECT "t"."tx_id", "w"."task_id"
FROM tx_rows AS "t" JOIN rewards AS "w" ON "w"."row_n" = "t"."row_n"
RETURNING md.user_rewards."id"
```

**Comment by houtn11** [comment, marked as answer]
`VALUES` is not supported, but there is a workaround - https://github.com/go-jet/jet/issues/69#issuecomment-761809025
